src/pages/create-course/CreateCourse.tsx

```diff
@@ -76,6 +77,29 @@ const CreateCourse = () => {
     handleNonInputValueChange('sections', value)
   }
 
+  const handleSectionInputChange = (
+    id: number,
+    field: keyof CourseSection,
+    value: string
+  ) => {
+    const sectionToEdit = data.sections.find((section) => section.id === id)
+    if (sectionToEdit) {
```

Comment:
Also can be simplified 

`sectionToEdit && Object.defineProperty(sectionToEdit, field, { value })`